internal/catalog/internal/controllers/failover/controller_test.go

```diff
@@ -0,0 +1,251 @@
+		Write(suite.T(), suite.client)
+
+	suite.client.WaitForStatusCondition(suite.T(), failover.Id, StatusKey, ConditionMissingService)
+
+	// Provide the service.
+	apiServiceData := &pbcatalog.Service{
+		Workloads: &pbcatalog.WorkloadSelector{Prefixes: []string{"api-"}},
+		Ports: []*pbcatalog.ServicePort{{
+			TargetPort: "http",
+			Protocol:   pbcatalog.Protocol_PROTOCOL_HTTP,
+		}},
+	}
+	_ = rtest.Resource(types.ServiceType, "api").
+		WithData(suite.T(), apiServiceData).
+		Write(suite.T(), suite.client)
+	suite.client.WaitForStatusCondition(suite.T(), failover.Id, StatusKey, ConditionOK)
+
+	// Update the failover to reference an unknown port
+	failoverData = &pbcatalog.FailoverPolicy{
+		PortConfigs: map[string]*pbcatalog.FailoverConfig{
+			"http": {
+				Destinations: []*pbcatalog.FailoverDestination{{
+					Ref:  apiServiceRef,
+					Port: "http",
+				}},
+			},
+			"admin": {
+				Destinations: []*pbcatalog.FailoverDestination{{
+					Ref:  apiServiceRef,
+					Port: "admin",
+				}},
+			},
+		},
+	}
+	_ = rtest.Resource(types.FailoverPolicyType, "api").
+		WithData(suite.T(), failoverData).
+		Write(suite.T(), suite.client)
+	suite.client.WaitForStatusCondition(suite.T(), failover.Id, StatusKey, ConditionUnknownPort("admin"))
+
+	// update the service to fix the stray reference
+	apiServiceData = &pbcatalog.Service{
+		Workloads: &pbcatalog.WorkloadSelector{Prefixes: []string{"api-"}},
+		Ports: []*pbcatalog.ServicePort{
+			{
+				TargetPort: "http",
+				Protocol:   pbcatalog.Protocol_PROTOCOL_HTTP,
+			},
+			{
+				TargetPort: "admin",
+				Protocol:   pbcatalog.Protocol_PROTOCOL_HTTP,
+			},
+		},
+	}
+	_ = rtest.Resource(types.ServiceType, "api").
+		WithData(suite.T(), apiServiceData).
+		Write(suite.T(), suite.client)
+	suite.client.WaitForStatusCondition(suite.T(), failover.Id, StatusKey, ConditionOK)
+
+	// change failover leg to point to missing service
+	failoverData = &pbcatalog.FailoverPolicy{
+		PortConfigs: map[string]*pbcatalog.FailoverConfig{
+			"http": {
+				Destinations: []*pbcatalog.FailoverDestination{{
+					Ref:  apiServiceRef,
+					Port: "http",
+				}},
+			},
+			"admin": {
+				Destinations: []*pbcatalog.FailoverDestination{{
+					Ref:  otherServiceRef,
+					Port: "admin",
+				}},
+			},
+		},
+	}
+	_ = rtest.Resource(types.FailoverPolicyType, "api").
+		WithData(suite.T(), failoverData).
+		Write(suite.T(), suite.client)
+	suite.client.WaitForStatusCondition(suite.T(), failover.Id, StatusKey, ConditionMissingDestinationService(otherServiceRef))
+
+	// Create the missing service, but forget the port.
+	otherServiceData := &pbcatalog.Service{
+		Workloads: &pbcatalog.WorkloadSelector{Prefixes: []string{"other-"}},
+		Ports: []*pbcatalog.ServicePort{{
+			TargetPort: "http",
+			Protocol:   pbcatalog.Protocol_PROTOCOL_HTTP,
+		}},
+	}
+	_ = rtest.Resource(types.ServiceType, "other").
+		WithData(suite.T(), otherServiceData).
+		Write(suite.T(), suite.client)
+	time.Sleep(1 * time.Second)
```

Comment:
Is this sleep required? It feels like `WaitForStatusCondition` should handle this.

Reply:
It's a stray remnant of development. Removed.